rust/cardano-chain-follower/src/metadata/cip509/mod.rs

```diff
@@ -591,9 +591,14 @@ impl Cip509 {
                                                             return None;
                                                         },
                                                     };
-                                                    // Extract the CIP19 hash and push into array
-                                                    if let Some(h) = extract_cip19_hash(&addr) {
-                                                        pk_addrs.push(h);
+                                                    // Only extract the CIP19 hash when it is a
+                                                    // stake address
+                                                    if addr.contains("stake") {
```

Comment:
Its theoretically possible for the address to encode the letters `stake` internally.  You should check the prefix explicitly.